data-reconciliation-app/docs/adr/01-data-ingest.md

```diff
@@ -2,126 +2,149 @@
-  - csv will have 2 columns:
-    - one unique identifier (string)
-    - one attribute associated with the unique identifier (string)
+- Members of the network will decide on a data schema ahead of ingest.
+- Members and Users of the network can ingest data via the API. Data will be ingested via JSON which will include:
+  - one unique identifier (string)
+  - one attribute associated with the unique identifier (string)
 - There is likely going to be a UI on top of our ingest API...so we will treat our app like a backend system. Therefore, we will accept JSON into our ingest API.
 
 ## Decision
 
-## Model for K-V store
+## API
 
-```
-//psuedo code for model based off forum app
+### Endpoint
 
-// this is unique for each user
-type User = string;
+- Description
+  - Ingest member or user data into our application and add data to K-V store
+- Path
+  - /ingest
+- HTTP Method
+  - POST
+- URL Params
+  - N/A
+- Headers
+  - content-type: application/json
+- Request will be a `ccfapp.Request Object`. We will interrogate the `ccfapp.Request Object` to extract the user or member and [authenticate them via certficates](#security). We will read the request body as a JSON.
+- API Status Codes
+  - OK
+    - Status: 200
+    - Status Text: "Votes have been successfully recorded"
+  - UNAUTHORIZED
+    - Status: 401
+    - Status Text: "Unauthorized"
+  - BAD_REQUEST
+    - Status: 400
+    - Status Text: "Validation Error"
 
-interface AttributeBase {
-  votes: Record<User, T>;
-}
+### Security
 
-interface StringAttribute extends AttributeBase<string> {
-  type: "string";
-}
+Users will be authenticated via certificates, which is natively supported by the CCF framework.
+
+//TODO Add reference to this future ADR: https://github.com/microsoft/ccf-app-samples/issues/102
+
+## Models
 
-interface NumericAttribute extends AttributeBase<number> {
-  type: "number";
+### Data Schema Model
+
+The ingest model has been documented by [this ADR](./02-data-schema-strategy.md#option3-defining-data-schema-via-deployment). Our ingest data model will look like:
+
+```
+export interface DataSchema {
+    key: string;
+    value: string | number;
 }
+```
+
+### Data Record Model
 
-type Attribute = StringAttribute | NumericAttribute;
+Upon ingest, `DataSchema` records will be mapped to a `DataRecord` model. `DataRecord`s can be string or numeric.
 
-type AttributeMap = ccfapp.TypedKvMap<string, Attribute>;
 ```
+export type DataAttributeType = string | number;
+export interface DataRecordProps {
+  key: string;
+  value: DataAttributeType;
+}
+```
+
+### Repository Model
+
+Our repository will be a K-V store and defined by the `ReconciledRecord` model. A `ReconciledRecord` represents all of the `DataRecords` across the network and each member/users' opinion of the record.
 
-### Sample Ingest Data
+The `ReconciledRecord` key will be the unique id for the data record and the `ReconciliationMap` will be a map of each member/user ID and their opinion of the record.
```

Comment:
I don't see the KV Store as the `ReconciledRecord` model. I see it as the storage in which we execute Reconciliation on the data.  The `ReconciledRecord` is dinamically created when a User requests for reconciliation. Am I wrong?

The way I see the project is that we:
1. ingest data (N users ingesting)
2. store them in "central" data repository for all users (KV Store)
3. perform reconciliation on this repository having the "requester" as reference (N users requesting)

Reply:
If you look at kv-repository.ts lines 123-127, the store is a key and value, where the value is a ReconciledRecord.

"The ReconciledRecord is dynamically created when a User requests for reconciliation" - Depending on the reporting design, when the user requests for reconciliation, we will pull from K-V store, do reconciliation logic and dynamically create a SummaryResults to return to the user

Reply:
I updated some language @julioalex-rezende lmk if it makes more clear

Reply:
The way I saw is that the SummaryResult would be the reconciled record. The KV store is just the repository from which we will query data, but the data there is not reconciled, only grouped.

But I understood the idea. I've just thought in a different way. Thank you for clarifying :)

Reply:
This thinking is correct! I think the confusion lies in the name of the actual repository record, but in terms of what object does what, you have that clear